src/utils/Logging.ts

```diff
@@ -703,67 +704,70 @@ export default class Logging {
       log.tenantID = Constants.DEFAULT_TENANT;
     }
     // Log
-    await LoggingStorage.saveLog(log.tenantID, log);
+    const logID = await LoggingStorage.saveLog(log.tenantID, log);
     // Log in Cloud Foundry
     if (Configuration.isCloudFoundry()) {
       // Bind to express app
       CFLog.logMessage(Logging.getCFLogLevel(log.level), log.message);
     }
+
+    return logID;
   }
 
-  private static anonymizeSensitiveData(message: any): void {
+  private static anonymizeSensitiveData(message: any): any {
     if (!message || typeof message === 'number' || Utils.isBoolean(message) || typeof message === 'function') {
-      // eslint-disable-next-line no-useless-return
-      return;
-    } else if (typeof message === 'string') {
+      return message;
+    } else if (typeof message === 'string') { // If the message is a string
+      // Check if it is a query string
+      const dataParts: string[] = message.split('&');
+      if (dataParts.length > 1) {
+        for (let i = 0; i < dataParts.length; i++) {
+          const dataPart = dataParts[i];
+          for (const sensitiveData of Constants.SENSITIVE_DATA) {
+            if (dataPart.toLowerCase().startsWith(sensitiveData.toLocaleLowerCase())) {
+              // Anonymize each query string part
+              dataParts[i] = dataPart.substring(0, sensitiveData.length + 1) + Constants.ANONYMIZED_VALUE;
+            }
+          }
+        }
+        message = dataParts.join('&');
+        return message;
+      }
+      // If it is a string, but not a query string
       for (const sensitiveData of Constants.SENSITIVE_DATA) {
-        // Anonymize
-        message.replace(new RegExp(sensitiveData, 'gi'), Constants.ANONYMIZED_VALUE);
+        if (new RegExp(sensitiveData + '[:=](.*)', 'gi').test(message)) { // If it contains sensitive data and it matches the RegExp, Anonymize
+          message = message.substring(0,sensitiveData.length + 1) + Constants.ANONYMIZED_VALUE;
```

Comment:
Missing space after comma. 

Please install the vscode plugin recommended for the repo and give them the rights they need to format and lint the code added.

Reply:
I have all extensions installed, I have to check what permissions are missing.